execution_engine/src/core/engine_state/upgrade.rs

```diff
@@ -224,20 +237,46 @@ where
         Ok(())
     }
 
-    /// Store new system contract.
-    fn store_contract(
+    /// Check if a contract's entry points differ from a given set.
+    fn has_different_entry_points(
+        &self,
+        contract: &Contract,
+        entry_points_to_check: &EntryPoints,
+    ) -> bool {
+        for new_entry_point in entry_points_to_check.clone().take_entry_points().iter() {
+            match contract.entry_points().get(new_entry_point.name()) {
+                None => {
+                    // An entry point has been added.
+                    return true;
+                }
+                Some(old_entry_point) => {
+                    // An existing entry point has been modified.
+                    if !new_entry_point.eq(old_entry_point) {
+                        return true;
+                    }
+                }
+            }
+        }
+        // Return true when Entry points have been removed.
+        !contract
+            .entry_points()
+            .keys()
+            .eq(entry_points_to_check.keys())
```

Comment:
Since we already check that all `entry_points_to_check` exist in `contract.entry_points()` then it should be enough to compare lengths `contract.entry_points().len() == entry_points.to_check.keys().len()`.

Reply:
Keeping it as is, because `keys()` returns an iterator, so the `len` method isn't present

Reply:
But `len()` should be available on `contract.entry_points()`.

Reply:
``` no method named `len` found for reference `&EntryPoints` in the current scope```

Reply:
`EntryPoints` is our type though and it's just a wrapper around `BTreeMap`, so we can add it. If it's equivalent to the current version then why not use the simpler and faster approach?